src/utils/merge-source-maps.js

```diff
@@ -0,0 +1,79 @@
+const { encode, decode } = require('sourcemap-codec');
+
+function traceSegment (map, mappings, line, column, name) {
+  const segments = mappings[line];
+  if (!segments) return null;
+
+  // binary search through segments for the given column
+  let i = 0;
+  let j = segments.length - 1;
+
+  while (i <= j) {
+    const m = (i + j) >> 1;
```

Comment:
Is this bitshift basically mean divide by 2?

This binary search looks reminds me of an interview question. Never thought I would see on in real code 😆

Reply:
Although good call not using the recursive solution 😉

Reply:
Oh, I absolutely did not write this code!

It's a reworking of Rollup's collapse source maps routine at https://github.com/rollup/rollup/blob/master/src/utils/collapseSourcemaps.ts, which I've used quite a bit.